src/edu/washington/escience/myria/api/encoding/QueryConstruct.java

```diff
@@ -97,6 +97,7 @@
        * setting the threshold to be the number of operators is enough.
        */
     }
+    Preconditions.checkArgument(loopCount < allOperators.size());
```

Comment:
If the preceding loop terminates when `loopCount == allOperators.size()`, won't the post-increment violate this invariant (actually I think it will make `loopCount == allOperators.size()+1`)? Also note that `setConsumerSchema()` could be called `allOperators.size()+1` times, since the second part of the conjunction is evaluated after the first part. This could be fixed by just swapping the two parts of the conjunction to force short-circuiting.